components/merkledb/src/group.rs

```diff
@@ -106,11 +107,31 @@ where
     }
 }
 
+impl<T, K, I> Group<T, K, I>
+where
+    T: RawAccess,
+    K: BinaryKey + ?Sized,
+    I: FromAccess<T>,
+{
+    /// Returns an iterator over the keys from this group.
+    ///
+    /// **Note:** use this method carefully, because storing all keys in memory may
```

Comment:
I'd say "avoid calling `collect` on the returned iterator, since ...", because obtaining iterator itself doesn't consume a large amount of RAM (or does it?).

Reply:
We consume RAM before obtaining iterator when we call `suffixes` method.

Reply:
Then I think that the note should be expanded with explanation why exactly RAM is consumed, since it's not really obvious -- usually, creating an iterator is a cheap operation.